gateway/reverse_proxy.go

```diff
@@ -829,7 +821,24 @@ func (rt *TykRoundTripper) RoundTrip(r *http.Request) (*http.Response, error) {
 		return rt.h2ctransport.RoundTrip(r)
 	}
 
-	return rt.transport.RoundTrip(r)
+	apiSpec := r.Context().Value("apiSpec").(*APISpec)
+	r = r.WithContext(context.WithValue(r.Context(), "apiSpec", nil))
```

Comment:
We can make APISpec param of TykRoundTripper, we have access to it when creating this object: https://github.com/TykTechnologies/tyk/blob/ca30b8fd7f9743f7100284687c2d7721306be18b/gateway/reverse_proxy.go#L740

Reply:
@buger I thought if that, but TykRoundTripper is returned from httpTransport method which is called once if transport is not created, so ApiSpec won t get updated?